ngraph/core/reference/include/ngraph/runtime/reference/region_yolo.hpp

```diff
@@ -0,0 +1,175 @@
+//*****************************************************************************
+// Copyright 2017-2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+
+#include <cmath>
+
+#include "ngraph/shape.hpp"
+
+namespace ngraph
+{
+    namespace runtime
+    {
+        namespace reference
+        {
+            static inline int entry_index(int width,
+                                          int height,
+                                          int coords,
+                                          int classes,
+                                          int outputs,
+                                          int batch,
+                                          int location,
+                                          int entry)
+            {
+                int n = location / (width * height);
+                int loc = location % (width * height);
+                return batch * outputs + n * width * height * (coords + classes + 1) +
+                       entry * width * height + loc;
+            }
+
+            template <typename T>
+            static inline T sigmoid(float x)
+            {
+                return static_cast<T>(1.f / (1.f + std::exp(-x)));
+            }
+            template <typename T>
+            static inline void softmax_generic(
+                const T* src_data, T* dst_data, int batches, int channels, int height, int width)
+            {
+                int start = 0;
+                for (unsigned int b = 0; b < batches; b++)
+                {
+                    for (unsigned int i = start; i < height * width; i++)
+                    {
+                        T max = src_data[b * channels * height * width + i];
+                        for (unsigned int c = 0; c < channels; c++)
+                        {
+                            T val =
+                                src_data[b * channels * height * width + c * height * width + i];
```

Comment:
So i see that this `b * channels * height * width + c * height * width + i` is used multiple times here, maybe introducing some base offset at least for `b * channels * height * width` would be a good idea?

Reply:
Agree. I added offset and area variables for these values.